sympy/concrete/expr_with_limits.py

```diff
@@ -25,12 +26,12 @@ def _process_limits(*symbols):
     limits = []
     orientation = 1
     for V in symbols:
-        if isinstance(V, Symbol):
+        if isinstance(V, (Symbol, Idx)):
```

Comment:
```
>>> i = Idx('i', range=(1, 4))
>>> A = IndexedBase('A')
>>> Sum(A[i], (i, 1, 2))
```
That's an opinion. `i` contains a range, but you specify another range. Just consider that allowing this behavior, then revoking it, would constitute a compatibility break, whereas the other way round would not.